app/lib/search/text_utils.dart

```diff
@@ -33,3 +34,12 @@ Iterable<String> splitForIndexing(String text) {
   if (text == null || text.isEmpty) return new Iterable.empty();
   return text.split(_separators).where((s) => s.isNotEmpty);
 }
+
+List<String> extractExactPhrases(String text) => _exactTermRegExp
+    .allMatches(text)
```

Comment:
For a search string like:
```
before "hello world" middle "greet from world" after
```
This will probably give 3 matches:
```
hello world
 middle 
greet from world
```

I think it would be easier to not use regexp in the first place and just change the tokenization mechanism of the text.

Reply:
Something like that is addressed in this test: https://github.com/dart-lang/pub-dartlang-dart/pull/579/files#diff-fdba8841ec20cef640d3f49523edcaf9R34

I think a `split` or `indexOf`-based phrase extraction will be more complex than the regexp. I'll change to the non-greedy version though, it may be faster.